src/kernels/compute_second_function.cl

```diff
@@ -1,48 +1,129 @@
 #include "common.h"
 
+int3_union get_nearby_point(int offset, int dim, int3_union point)
+{
+	int3_union nearby = point;
+	nearby.elem[dim] += offset;
+	return nearby;	
+}
+
+
 kernel void
 compute_second_function(
 	global const T* zeta,
-	const T3_union delta,
+	global const T* delta,
```

Comment:
Why not `T3_union`? In the test code you pass this argument as OpenCL buffer, and in the main code you pass it as `Vec3` (which is a struct). The main code will not work that way. Please, use `T3_union` and pass `Vec3` in the test code.